### PR TITLE
Deploy only compressed metadata for dotnet sdk implementation

### PR DESCRIPTION
**Pull request by KevinRansom**
When we deploy fsharp.core and fsharp.compiler.service to nuget we deploy uncompressed metadata because we need the LTS dotnet sdk to be usable in building apps consuming them.  The current LTS does not understand compressed F# metadata.

However, the implementation F# compiler in the latest dotnet SDK understands compressed F# metadata so we don't need to waste those compressable bytes.

With this PR ... the dotnet SDK should be almost 3 MBytes smaller on disk and in VM, which has to be a good thing.

### CODE REVIEW

**Comment by vzarytovskii** [comment]
What concerns me is that people update fsharp.core quite often without updating sdk, which may result in either cryptic errors or undefined behavior.

**Comment by vzarytovskii** [comment]
> > What concerns me is that people update fsharp.core quite often without updating sdk, which may result in either cryptic errors or undefined behavior.
> 
> @vzarytovskii the nuget package that we embed in the sdk FSharp\library-packs folder and ship to nuget still has uncompressed metadata.
> 
> This change only impacts the files that form the compiler implementation.

OH, got it now

File: src/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.fsproj [hunk 1]
```diff
@@ -34,8 +34,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(MSBuildThisFileDirectory)..\fsi\fsiProject\fsi.fsproj" />
-    <ProjectReference Include="$(MSBuildThisFileDirectory)..\fsc\fscProject\fsc.fsproj" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\fsi\fsiProject\fsi.fsproj">
+        <AdditionalProperties>Configuration=ReleaseCompressed;CompressAllMetadata=true</AdditionalProperties>
```
Comment (T-Gro) [inline src/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.fsproj:38]: Wouldn't the property setting `CompressAllMetadata=true` be sufficient?
Or is there another reason for having a separate Configuration I am not seeing?

**Review by T-Gro — COMMENTED** [review]
Is the new separate Configuration a mechanism of isolation, or does it do anything special?
I can see the `CompressAllMetadata=true` doing the real work, is ReleaseCompressed there to ensure it has its own dedicated artefacts subfolder?

File: src/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.fsproj (hunk 1 above)
Comment (KevinRansom) [reply src/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.fsproj:38]: No, fsc and fsi, have the same set of projects as dependencies as appear later on.  The later on set need to be either release or debug, with the standard compressions.  The reason for this is to ensure that the fsi and fsc that we embed in the sdk have the correct (I.e. compressed state, and the fsharp.compiler.service and fsharp.core nuget packages that are also embedded also have the correct, I.e not compressed state.

**Comment by KevinRansom** [comment]
> Is the new separate Configuration a mechanism of isolation, or does it do anything special? I can see the `CompressAllMetadata=true` doing the real work, is ReleaseCompressed there to ensure it has its own dedicated artefacts subfolder?
Yes, quite so, it needs it's own folder, with it's separate bits.

**Review by T-Gro — APPROVED** [review]

File: src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Nuget.fsproj [hunk 2]
```diff
@@ -9,7 +9,8 @@
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <DefineConstants>$(DefineConstants);COMPILER</DefineConstants>
     <OtherFlags>$(OtherFlags) --warnon:1182</OtherFlags>
-    <Tailcalls>true</Tailcalls> <!-- .tail annotations always emitted for this binary, even in debug mode -->
```
Comment (abonie) [inline src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Nuget.fsproj:12]: Is this removed on purpose?

**Review by abonie — APPROVED** [review]

**Pull request merged**